src/js/actions/ui.js

```diff
@@ -257,6 +257,43 @@ define(function (require, exports) {
     updatePanelSizes.modal = true;
 
     /**
+     * Set the overlay offsets in PS in anticipation of opening/creating the
+     * first document (i.e., from a state in which there are no open documents).
+     * This is used, e.g., to ensure that the offsets account for the UI columns
+     * that will be shown once the document is open. See #1999 for more details.
+     *
+     * @return {Promise}
+     */
+    var setOverlayOffsetsForFirstDocument = function () {
+        var flux = this.flux,
+            applicationStore = flux.store("application");
+
+        if (applicationStore.getDocumentCount() > 0) {
+            return Promise.resolve();
+        }
+
+        var preferencesStore = flux.store("preferences"),
+            uiStore = flux.store("ui"),
+            preferences = preferencesStore.getState(),
+            columnCount = 0;
+
+        if (preferences.get(uiStore.components.LAYERS_LIBRARY_COL), true) {
```

Comment:
I amended the commit here and just below to use `true` as the default if the key is not found in the preferences table. This is how it works in `PanelSet`.